sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DeliveringDebugFeaturesInProductionTest.cs

```diff
@@ -37,41 +37,58 @@ public class DeliveringDebugFeaturesInProductionTest
-        }
+                additionalReferences: AdditionalReferencesNetCore2);
 
         [TestMethod]
         [TestCategory("Rule")]
-        public void DeliveringDebugFeaturesInProduction_CS_Disabled()
-        {
-            Verifier.VerifyNoIssueReported(@"TestCases\DeliveringDebugFeaturesInProduction.cs",
+        public void DeliveringDebugFeaturesInProduction_NetCore2_CS_Disabled() =>
+            Verifier.VerifyNoIssueReported(@"TestCases\DeliveringDebugFeaturesInProduction.NetCore2.cs",
                 new CSharp.DeliveringDebugFeaturesInProduction(),
-                additionalReferences: AdditionalReferences);
-        }
+                additionalReferences: AdditionalReferencesNetCore2);
 
         [TestMethod]
         [TestCategory("Rule")]
-        public void DeliveringDebugFeaturesInProduction_VB()
-        {
-            Verifier.VerifyAnalyzer(@"TestCases\DeliveringDebugFeaturesInProduction.vb",
+        public void DeliveringDebugFeaturesInProduction_NetCore2_VB() =>
+            Verifier.VerifyAnalyzer(@"TestCases\DeliveringDebugFeaturesInProduction.NetCore2.vb",
                 new VisualBasic.DeliveringDebugFeaturesInProduction(AnalyzerConfiguration.AlwaysEnabled),
-                additionalReferences: AdditionalReferences);
-        }
+                additionalReferences: AdditionalReferencesNetCore2);
 
         [TestMethod]
         [TestCategory("Rule")]
-        public void DeliveringDebugFeaturesInProduction_VB_Disabled()
-        {
-            Verifier.VerifyNoIssueReported(@"TestCases\DeliveringDebugFeaturesInProduction.vb",
+        public void DeliveringDebugFeaturesInProduction_NetCore2_VB_Disabled() =>
+            Verifier.VerifyNoIssueReported(@"TestCases\DeliveringDebugFeaturesInProduction.NetCore2.vb",
                 new VisualBasic.DeliveringDebugFeaturesInProduction(),
-                additionalReferences: AdditionalReferences);
-        }
+                additionalReferences: AdditionalReferencesNetCore2);
 
-        internal static IEnumerable<MetadataReference> AdditionalReferences =>
+#if NETCOREAPP
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void DeliveringDebugFeaturesInProduction_NetCore3_CS() =>
+            Verifier.VerifyAnalyzer(@"TestCases\DeliveringDebugFeaturesInProduction.NetCore3.cs",
+                new CSharp.DeliveringDebugFeaturesInProduction(AnalyzerConfiguration.AlwaysEnabled),
+                additionalReferences: AdditionalReferencesNetCore3);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void DeliveringDebugFeaturesInProduction_NetCore3_VB() =>
+            Verifier.VerifyAnalyzer(@"TestCases\DeliveringDebugFeaturesInProduction.NetCore3.vb",
+                new VisualBasic.DeliveringDebugFeaturesInProduction(AnalyzerConfiguration.AlwaysEnabled),
+                additionalReferences: AdditionalReferencesNetCore3);
+
+        internal static IEnumerable<MetadataReference> AdditionalReferencesNetCore3 =>
```

Comment:
This was actually misleading since we don't reference the target framework directly. In practice, this will load the libraries from the SDK version linked by the test project.